conda/base/context.py

```diff
@@ -2068,25 +2069,30 @@ def validate_prefix_name(prefix_name: str, ctx: Context, allow_base=True) -> str
     """Run various validations to make sure prefix_name is valid"""
     from ..exceptions import CondaValueError
 
-    disallowed = (
-        PREFIX_NAME_DISALLOWED_CHARS_WIN if on_win else PREFIX_NAME_DISALLOWED_CHARS
-    )
-    if disallowed.intersection(prefix_name):
-        if "%" in disallowed:
-            # This symbol causes formatting errors in the message below when raised.
-            # It needs to be escaped as a double %% in order to be rendered correctly.
-            # Raw strings didn't help.
-            disallowed.remove("%")
-            disallowed.add("%%")
+    if PREFIX_NAME_DISALLOWED_CHARS.intersection(prefix_name):
         raise CondaValueError(
             dals(
-                f"""
-                Invalid environment name: {prefix_name!r}
-                Characters not allowed: {disallowed}
+                """
+                Invalid environment name: %(prefix_name)r
+                Characters not allowed: %(disallowed)s
                 If you are specifying a path to an environment, the `-p`
                 flag should be used instead.
                 """
-            )
+            ),
+            prefix_name=prefix_name,
+            disallowed=PREFIX_NAME_DISALLOWED_CHARS,
+        )
+    if on_win and PREFIX_NAME_DISCOURAGED_CHARS_WIN.intersection(prefix_name):
+        warnings.warn(
+            dals(
+                f"""
+                Environment name {prefix_name!r}
+                contains characters from {PREFIX_NAME_DISCOURAGED_CHARS_WIN}
+                which are known to cause issues.
+                Future conda versions may disallow them in environment names.
```

Comment:
This is intentionally worded vaguely because the whole process on figuring out which/how many of those characters should be discouraged and when to implement it is still TBD.